patroni/config.py

```diff
@@ -45,7 +46,11 @@ def __init__(self, config_file=None, config_env=None):
         self._config_file = None if config_env else config_file
         self._modify_index = -1
         self._dynamic_configuration = {}
-        self._local_configuration = yaml.safe_load(config_env) if config_env else self._load_config_file()
+        if config_env:
+            self._local_configuration = yaml.safe_load(config_env)
+        else:
+            self.__environment_configuration = self._build_environment_configuration()
```

Comment:
No, it's not possible.
`_load_config_file` can be executed more than once, but `_build_environment_configuration` MUST be executed only once, because it removes all `PATRONI_` variables from `os.environ`. I did it intensionally, because in the future we may want to dump `self.__environment_configuration` into the file and **re-execute** Patroni (just to reset environment).